scania-digital-design-system/tegel
# fix(dropdown): empty no-result-text prop

**Describe pull-request**  
If the user uses lazy loading, there is a bug in which we display a "No results" message for a short time while user logic is fetching results. If the user sets noResultsText to an empty string now, it wont trigger no results message to pop up.
The second fix is to send an error in the console if no tds-dropdown option is available once users interact with the component.

**Solving issue**  

Fixes: [CDEP-2974](https://tegel.atlassian.net/browse/CDEP-2974)

**How to test**  
1. Go to the preview link.
2. Enable filtering
3. Try to input a query that will not match any data.
4. No result message should pop up.
5. Delete "No result text" from Storybook control.
6. Try to search again for a query that will not match any data.
7. The message "No result" should NOT pop up.
8. Branch on you local
9. Drop all <tds-dropdown-option> from story file.
10. Interact with component. Nothing should happen.
11. Take a look at to console, there should be an error message saying that the dropdown is missing data.

**Suggested test steps**




[CDEP-2947]: https://tegel.atlassian.net/browse/CDEP-2947?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CDEP-2974]: https://tegel.atlassian.net/browse/CDEP-2974?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
